tests/duet-song.spec.ts

```diff
@@ -0,0 +1,165 @@
+  test.fixme(browserName === 'firefox', 'Test fails super often on FF');
+  test.slow();
+  await page.goto('/?e2e-test');
+  await pages.landingPage.enterTheGame();
+
+  await test.step('Select computer`s mic', async () => {
+    await pages.inputSelectionPage.selectComputersMicrophone();
+    await pages.computersMicConnectionPage.continueToTheSong();
+  });
+
+  await test.step('Select song language', async () => {
+    await pages.mainMenuPage.goToSingSong();
+    await pages.songLanguagesPage.ensureSongLanguageIsSelected(spanishLang);
+    await pages.songLanguagesPage.continueAndGoToSongList();
+  });
+
+  await test.step('Go to duet playlist and pick up the song', async () => {
+    await pages.songListPage.goToPlaylist(duetsPlaylist);
+    await pages.songListPage.expectPlaylistToBeSelected(duetsPlaylist);
+    await expect(pages.songListPage.getDuetSongIcon(duetSpanSong)).toBeVisible();
+  });
+
+  await test.step('Check visibility of song language flag and open song', async () => {
+    await pages.songListPage.expectSongToBeMarkedWithLanguageFlagIcon(duetSpanSong, spanishLang);
+    await pages.songListPage.navigateToSongWithKeyboard(duetSpanSong);
+    await pages.songListPage.openPreviewForSong(duetSpanSong);
+  });
+
+  await test.step('Set Pass-The-Mic game mode', async () => {
+    await pages.songPreviewPage.navigateToGameModeSettingsWithKeyboard();
+    await pages.songPreviewPage.toggleGameMode(); // Duel
+    await pages.songPreviewPage.toggleGameMode(); // Pass the mic
+    await pages.songPreviewPage.expectGameModeToBe(gameMode);
+  });
+
+  await test.step('Enter player name and play the song with 1 player', async () => {
+    await pages.songPreviewPage.goNext();
+    await expect(pages.songPreviewPage.getPlayerNameInput(player1)).toBeVisible();
+    await pages.songPreviewPage.navigateAndEnterPlayerNameWithKeyboard(player1, player1Name);
+    await expect(pages.songPreviewPage.getPlayerNameInput(player2)).not.toBeVisible();
+    await pages.songPreviewPage.playTheSong();
+  });
+
+  await test.step('Artist and song title should be displayed', async () => {
+    await pages.gamePage.expectArtistAndSongTitleToBeShown(artist, songTitle);
+  });
+
+  await test.step('Song lyrics are enable just for player1', async () => {
+    await expect(pages.gamePage.getSongLyricsForPlayerElement(player1)).toBeVisible();
+    await expect(pages.gamePage.getSongLyricsForPlayerElement(player2)).not.toBeVisible();
+  });
+
+  await test.step('Score should be visible for player1', async () => {
+    await expect(pages.gamePage.getPlayerScoreElement(player1)).toBeVisible();
+    await expect(pages.gamePage.getPlayerScoreElement(player2)).not.toBeVisible();
+  });
+
+  await test.step('Pass the mic progress and change indicator icon should be displayed', async () => {
+    await expect(pages.gamePage.passTheMicProgressElement).toBeVisible();
+    // await expect(pages.gamePage.changeIndicatorIcon).toBeVisible({ timeout: 20_000 });
```

Comment:
lets create a separate issue to fix this

Reply:
Ok, done #207 :)